src/main/java/uk/ac/ebi/eva/evaseqcol/service/SeqColService.java

```diff
@@ -198,17 +202,25 @@ public List<String> fetchAndInsertAllSeqColByAssemblyAccession(
                     seqColStringListExtDataEntities, seqColIntegerListExtDataEntities, extendedNamesEntity.getNamingConvention()
                     );
 
-            Optional<String> seqColDigest = insertSeqColL1AndL2( // TODO: Check for possible self invocation problem
-                    levelOneEntity, seqColStringListExtDataEntities, seqColIntegerListExtDataEntities);
-            if (seqColDigest.isPresent()) {
-                logger.info(
-                        "Successfully inserted seqCol for assembly Accession " + assemblyAccession + " with naming convention " + extendedNamesEntity.getNamingConvention());
-                insertedSeqColDigests.add(seqColDigest.get());
-            } else {
-                logger.warn("Could not insert seqCol for assembly Accession " + assemblyAccession + " with naming convention " + extendedNamesEntity.getNamingConvention());
+            try {
+                Optional<String> seqColDigest = insertSeqColL1AndL2( // TODO: Check for possible self invocation problem
+                                                                     levelOneEntity, seqColStringListExtDataEntities, seqColIntegerListExtDataEntities);
+                if (seqColDigest.isPresent()) {
+                    logger.info(
+                            "Successfully inserted seqCol for assembly Accession " + assemblyAccession + " with naming convention " + extendedNamesEntity.getNamingConvention());
+                    InsertedSeqColEntity insertedSeqCol = new InsertedSeqColEntity(seqColDigest.get(), extendedNamesEntity.getNamingConvention().toString());
+                    ingestionResultEntity.addInsertedSeqCol(insertedSeqCol);
+                    ingestionResultEntity.incrementNumberOfInsertedSeqCols();
+                } else {
+                    logger.warn("Could not insert seqCol for assembly Accession " + assemblyAccession + " with naming convention " + extendedNamesEntity.getNamingConvention());
+                }
+            } catch (DuplicateSeqColException e) {
```

Comment:
Going along with what Tim was saying... if you catch this exception here it won't make it to the controller and give you the `HttpStatus.CONFLICT` code, which might not be what we want.